front/lib/plans/subscription.ts

```diff
@@ -162,7 +174,7 @@ export const internalSubscribeWorkspaceToFreeUpgradedPlan = async ({
 export const subscribeWorkspaceToPlan = async (
   auth: Authenticator,
   { planCode }: { planCode: string }
-): Promise<PlanType> => {
+): Promise<string | void> => {
```

Comment:
(We didn't really needed to return the PlanType cause we cannot swr the plan anyway)

Reply:
If we can I would still do it no?

Reply:
What is the string?

Reply:
It's the checkout URL.

I think an ideal typing here would be:

```
{
  plan: PlanType,
  checkoutUrl: string
}
```
?

Reply:
Agreed we wouldn't use it but from an API design standpoint that looks like a better API. You subscribe to a planCode the function gives you the plan and the checkout URL?

Reply:
Ah I see we don't have a type for a Plan independently of a subscription. Should we? We'll need it in poke no?

Assuming we had a type for an plan (independent of subscriptions), I would return that object to indicate which plan I'm about to subscribe to.

Reply:
We probably want to introduce a type to represent a `Plan` (in the sense of the model) and I would return one such object associated with `planCode`. 

We'll want to list plans in poke so we likely need that type right?

Naming wise maybe we rename PlanType SubscriptionType ? Maybe we keep PlanType as such (associated with a Workspace and representing really a Subscription) and introduce another name for the plan (but don't quite see which one)?

Reply:
Yes we could introduce a new type just for that but when you did subscribe to a free plan you want the plan with the subscription data.

I think the best would be to return void or just the checkoutUrl, but if we want the plan then I would go with b/ -> a PlanType to keep just one type in front, and a status to processing. So you know to which plan you are about to subscribe to, but you see it's processing and not active yet.

Reply:
Il fine with just an URL.

No processing state IMHO as it adds complexity where it should not. The subscription just does not exist yet.

I still think we’ll need that type in poke, no?

Reply:
Yes 👍

Reply:
Awesome, task here: https://github.com/dust-tt/tasks/issues/169